### PR TITLE
WIP: Add Halide recipe

### PR DESCRIPTION
**Pull request by arokem**

### CODE REVIEW

File: recipes/halide/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+#test:
+#  imports:
+#    - example
+#  commands:
+#    - python -c "import example;"
+
+about:
+  home: https://github.com/halide/Halide
+  license: BSD
```
Comment (ocefpaf) [inline recipes/halide/meta.yaml]: @arokem can you check which version of the BSD license? `BSD 2-Clause` or `BSD 3-Clause`?

File: recipes/halide/build.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# On OSX, we need to ensure we're using conda's gcc/g++
+if [[ `uname` == Darwin ]]; then
+     export CC=gcc
+     export LD=gcc
+     export CXX=g++
```
Comment (jakirkham) [inline recipes/halide/build.sh]: Why?

Comment (arokem) [inline recipes/halide/build.sh]: I am working my way forward slowly here, trying to emulate things I saw on https://github.com/conda-forge/staged-recipes/pull/220, but the truth is that I am not so sure how to go about it.

Comment (jakirkham) [inline recipes/halide/build.sh]: Does Halide use OpenMP? From this [search]( https://github.com/halide/Halide/search?utf8=%E2%9C%93&q=omp ), it looks like it generates code that does, but it itself doesn't. As such, I don't think we need the `gcc` package. So, try to follow this PR ( https://github.com/conda-forge/thrift-cpp-feedstock/pull/4 ) as an example.

Comment (arokem) [inline recipes/halide/build.sh]: I think that's right. Will look at the thrift recipe PR. 

For the time being, it looks like Halide requires a newer version of LLVM (3.5): Can't find llvm or version of llvm too old (we need 3.5 or greater):

Comment (jakirkham) [inline recipes/halide/build.sh]: @msarahan, any thoughts on a newer LLVM? We could also try build it from a recipe here, but I'm guessing that is going to take way too long.

**Pull request closed without merging**